ghc-prof-flamegraph.hs

```diff
@@ -9,35 +9,64 @@ import qualified Options.Applicative as Opts
 import qualified ProfFile as Prof
 
 data Options = Options
-  { optionsAlloc :: Bool
+  { optionsReportType :: ReportType
   } deriving (Eq, Show)
 
+
+data ReportType = Alloc   -- ^ Report allocations, percent
+                | Entries -- ^ Report entries, number
+                | Ticks   -- ^ Report ticks, number
+                | Bytes   -- ^ Report bytes allocated, number
+                deriving (Eq, Show)
+
 optionsParser :: Opts.Parser Options
 optionsParser = Options
-  <$> Opts.switch (Opts.long "alloc" <> Opts.help "Uses the allocation measurements instead of time measurements")
+  <$> (Opts.flag' Alloc (Opts.long "alloc" <> Opts.help "Uses the allocation measurements instead of time measurements")
+       Opts.<|> Opts.flag' Entries (Opts.long "entry" <> Opts.help "Usues entries the measurements instead of time measurements")
+       Opts.<|> Opts.flag' Bytes   (Opts.long "bytes" <> Opts.help "Memory measurements in bytes (+RTS -P -RTS)")
+       Opts.<|> Opts.flag Ticks Ticks (Opts.long "ticks" <> Opts.help "Time measurements in ticks (+RTS -P -RTS)"))
 
-generateFrames :: Options -> [Prof.Line] -> [String]
-generateFrames options lines0 =
-  let (entries, frames) = go [] lines0
-      unknown = 1000 - entries
+checkNames :: ReportType -> [String] -> Maybe String
+checkNames Alloc   _ = Nothing
+checkNames Entries _ = Nothing
+checkNames Ticks   n
+  | "ticks" `elem` n = Nothing
+  | otherwise        = Just "No ticks information, please run program with +RTS -P"
+checkNames Bytes n
+  | "bytes" `elem` n = Nothing
+  | otherwise        = Just "No ticks information, please run program with +RTS -P"
+
+normalize :: ReportType -> Double -> Int
+normalize Alloc = round . (10 *)
```

Comment:
Aha, sorry, I missed that! Let's leave it as-is, I do not recall right now why this was needed. I should have added a comment then!